File: leetcode/1347. Minimum Number of Steps to Make Two Strings Anagram/main.ex
defmodule Solution do
  @spec min_steps(s :: String.t(), t :: String.t()) :: integer
  def min_steps(s, t) do
    s_freq = Enum.reduce(String.to_charlist(s), init_freq(), &update_freq/2)
    t_freq = Enum.reduce(String.to_charlist(t), init_freq(), &update_freq/2)

    steps =
      Enum.reduce(0..25, 0, fn i, acc -> acc + abs(Enum.at(s_freq, i) - Enum.at(t_freq, i)) end)

    div(steps, 2)
  end

  @spec init_freq() :: list()
  defp init_freq, do: Enum.map(0..25, fn _ -> 0 end)

  @spec update_freq(char(), list()) :: list()
  defp update_freq(c, freq) do
    idx = c - 97
    List.replace_at(freq, idx, Enum.at(freq, idx) + 1)
  end
end

File: leetcode/1347. Minimum Number of Steps to Make Two Strings Anagram/main.ts
const minSteps = (s: string, t: string): number => {
  const [sFreq, tFreq] = [
    new Array<number>(26).fill(0),
    new Array<number>(26).fill(0),
  ]

  for (let i = 0; i < s.length; i++) {
    sFreq[s.charCodeAt(i) - "a".charCodeAt(0)]++
    tFreq[t.charCodeAt(i) - "a".charCodeAt(0)]++
  }

  let steps = 0
  for (let i = 0; i < 26; i++) {
    steps += Math.abs(sFreq[i] - tFreq[i])
  }

  return Math.floor(steps / 2)
}
